### PR TITLE
Rewrite psi check with xpaths instead of css selectors (fixes #84)

### LINKED ISSUE #84: PageSpeed Insights CI check failed
image

### PR DESCRIPTION
**Pull request by MyIsaak**
This PR rewrites the PSI check, specifically the class based selectors into xpath, because PSI now obfusticates their classes. This also makes it harder to select tags, however with xpath, one can select tags with text inside them, avoiding the use of classes. 

This PR fixes the broken builds, however the quality of the PDF is poor. Only the core web vitals are visable. Suggested fix would be to include a auto scroll down function in `utils.js`, to trick the page into loading all components, thus the PDF will contain all parts.

### CODE REVIEW

**Comment by MyIsaak** [comment]
Another idea is to set the viewport to null, which then sizes the viewport to the document, instead of a fixed value.

**Comment by DanielRuf** [comment]
> Another idea is to set the viewport to null, which then sizes the viewport to the document, instead of a fixed value.

Oh, that's interesting. Never knew or tried that =)

**Comment by MyIsaak** [comment]
I'm not sure if it works though. I'll make a quick test and see.

**Comment by MyIsaak** [comment]
Resizing the viewport didn't work. On the other hand, adding the scrolling function did, which I've now added. The PDF's quality as improved, because now we can see all components.

**Comment by MyIsaak** [comment]
@DanielRuf Do you think its now mergable?

File: modules/checks/psi.js [hunk 1]
```diff
@@ -2,18 +2,24 @@
...
 module.exports = async () => {
   if (no_cli_flags || options_keys.includes('--psi')) {
     const name = 'PageSpeed Insights'
+    /** @param {import("puppeteer").Page} page */
     async function tryBlock(page) {
       await page._client.send('Emulation.clearDeviceMetricsOverride')
       await retry(() => page.goto('https://developers.google.com/speed/pagespeed/insights/?url=' + url + '&tab=mobile'), 1000)
-      await page.waitForSelector('#page-speed-insights .pagespeed-results .result-tabs', { timeout: 60000 })
-      await page.click('#page-speed-insights .pagespeed-results .result-tabs .goog-tab:nth-child(1)')
+      await page.waitForFunction(() => document.body.innerText.includes("Discover what your real users are experiencing"), { timeout: 600000 })
```
Comment (DanielRuf) [inline modules/checks/psi.js:15]: Quite high timeout but ok for me.

**Comment by DanielRuf** [comment]
> @DanielRuf Do you think its now mergable?

Can be merged.

**Pull request merged**